### PR TITLE
TST: avoid matplotlib warnings in tests

### PR DESCRIPTION
**Pull request by jbrockmendel**
In master we get the following warnings in this file:

```
$ pytest pandas/tests/plotting/test_frame.py --skip-slow
[...]

pandas/tests/plotting/test_frame.py::TestDataFramePlots::test_mpl2_color_cycle_str
  pandas/pandas/plotting/_style.py:60: MatplotlibDeprecationWarning: Support for uppercase single-letter colors is deprecated since Matplotlib 3.1 and will be removed in 3.3; please use lowercase instead.
    [conv.to_rgba(c) for c in colors]

pandas/tests/plotting/test_frame.py::TestDataFramePlots::test_subplots_timeseries_y_axis
  pandas/pandas/plotting/_core.py:384: FutureWarning: Converting timezone-aware DatetimeArray to timezone-naive ndarray with 'datetime64[ns]' dtype. In the future, this will return an ndarray with 'object' dtype where each element is a 'pandas.Timestamp' with the correct 'tz'.
  	To accept the future behavior, pass 'dtype=object'.
  	To keep the old behavior, pass 'dtype="datetime64[ns]"'.
    numeric_data[col] = np.asarray(numeric_data[col])

pandas/tests/plotting/test_frame.py::TestDataFramePlots::test_line_area_stacked
pandas/tests/plotting/test_frame.py::TestDataFramePlots::test_line_area_stacked
pandas/tests/plotting/test_frame.py::TestDataFramePlots::test_line_area_stacked
pandas/tests/plotting/test_frame.py::TestDataFramePlots::test_line_area_stacked
pandas/tests/plotting/test_frame.py::TestDataFramePlots::test_line_area_stacked
pandas/tests/plotting/test_frame.py::TestDataFramePlots::test_line_area_stacked
pandas/tests/plotting/test_frame.py::TestDataFramePlots::test_line_area_stacked
pandas/tests/plotting/test_frame.py::TestDataFramePlots::test_line_area_stacked
pandas/tests/plotting/test_frame.py::TestDataFramePlots::test_line_area_stacked
pandas/tests/plotting/test_frame.py::TestDataFramePlots::test_line_area_stacked
pandas/tests/plotting/test_frame.py::TestDataFramePlots::test_line_area_stacked
pandas/tests/plotting/test_frame.py::TestDataFramePlots::test_line_area_stacked
pandas/tests/plotting/test_frame.py::TestDataFramePlots::test_line_area_stacked
pandas/tests/plotting/test_frame.py::TestDataFramePlots::test_line_area_stacked
pandas/tests/plotting/test_frame.py::TestDataFramePlots::test_line_area_stacked
pandas/tests/plotting/test_frame.py::TestDataFramePlots::test_line_area_stacked
  pandas/pandas/plotting/_core.py:1014: UserWarning: Attempted to set non-positive left xlim on a log-scaled axis.
  Invalid limit will be ignored.
    ax.set_xlim(left, right)
```

This PR addresses first and third of these.  The middle one is the topic of #26676.

### CODE REVIEW

**Comment by alimcmaster1** [comment]
Interesting I saw the same error in my PR:
```
   raise_assert_detail(obj, '{0} length are different'.format(obj),
E   AssertionError: Iterable are different
E   
E   Iterable length are different
E   [left]:  133
E   [right]: 128
E   [diff]: ['e', 'msg', 'missing_dependencies', 'dependency', 'hard_dependencies']
```
https://dev.azure.com/pandas-dev/pandas/_build/results?buildId=12339

Might raise an issue?

File: pandas/tests/plotting/test_frame.py [hunk 1]
```diff
@@ -166,7 +166,7 @@ def test_integer_array_plot(self):
 
     def test_mpl2_color_cycle_str(self):
         # GH 15516
-        colors = ['C' + str(x) for x in range(10)]
+        colors = ['c' + str(x) for x in range(10)]
```
Comment (QuLogic) [inline pandas/tests/plotting/test_frame.py]: Wait, is this really what raises the warning? `C0`, etc. is correct, and `c0` should not work (assuming these values are directly passed to Matplotlib.) Is Pandas doing some other processing of these colour names?

Comment (QuLogic) [reply pandas/tests/plotting/test_frame.py]: OK, looking at `_get_standard_colors`, I see Pandas does do extra changes, it checks the individual elements of both `[colors]` and `list(colors)`. The latter changes `'C0'` to `['C', '0']`, neither of which are valid.

I'm pretty sure this test was correct and it is in _that_ function that a warning capture context should be added.

Comment (jbrockmendel) [reply pandas/tests/plotting/test_frame.py]: @jorisvandenbossche any thoughts on this?  I think you introduced this test a while back.

Comment (jbrockmendel) [reply pandas/tests/plotting/test_frame.py]: Reverted this change, can address separately.

**Comment by jbrockmendel** [comment]
@alimcmaster1 is the issue you mentioned affecting all PRs or just these two?

File: pandas/tests/plotting/test_frame.py (hunk 1 above)
Comment (jorisvandenbossche) [reply pandas/tests/plotting/test_frame.py]: Didn't check in detail, but as @QuLogic says, this test is certainly testing something valuable, that `'C0'`, `'C1'`, etc color names are working, and should not be changed like the above.

**Comment by jorisvandenbossche** [comment]
> @alimcmaster1 is the issue you mentioned affecting all PRs or just these two?

It was a temporary failure (due to me merging something that should not have been merged) that is fixed by now.

**Review by jorisvandenbossche — APPROVED** [review]

**Comment by TomAugspurger** [comment]
Thanks!

**Pull request merged**